R/dependecies.R

```diff
@@ -0,0 +1,510 @@
+#' Get dependencies for all elements in lock
+get_ordered_dependencies <- function(lock) {
+   cat(crayon::blue(paste("Retrieving dependency info...")))
+   deps <- get_dependencies_for_list(lock, lock, list())
```

Comment:
Starts off the recursive chain.  Lock must be always available in it's original form, so that we know what we started out with.  The other option is to parse the list each time and look for !is_dependency_package, which should actually work fine, too.